### ISSUE TITLE
Can't translate current recurring state

### ISSUE THREAD

**Issue by Neui** [getting-things-gnome/gtg#490]
After setting a task to be recurring, the current state can't be translated (and is "ugly" in the sense that it says "other-day" instead of like "other day").

To allow flexibility for other languages, maybe use more specific strings such as "Every other day", "Every {day}. of the Month", "Every {weekday}", ...)

![It says "every other-day"](https://user-images.githubusercontent.com/1196130/96499693-083e9400-124e-11eb-8ff3-4ee60c362097.png) ![It says "Every 09 of the Month"](https://user-images.githubusercontent.com/1196130/96500092-a3d00480-124e-11eb-985f-6f7891b0465e.png)

**Comment by diegogangl** [comment]
Seems like we need to move the "Every" prefix into the strings themselves, and build a special string for other-day. What do you think @zeddo123 ?

**Comment by zeddo123** [comment]
> (and is "ugly" in the sense that it says "other-day" instead of like "other day")

Yeah, I know. It is not even easy to write but I had to make it that way as it would not be parsed correctly in the quick add entry otherwise.

>To allow flexibility for other languages, maybe use more specific strings such as "Every other day", "Every {day}. of the Month", "Every {weekday}", ...)

We are currently doing similar formatting here. Could you point me to the problem? (I don't know how translating works).
https://github.com/getting-things-gnome/gtg/blob/024029129c267fbaa82a74660e1ed7cb2d7b98ba/GTG/gtk/editor/recurring_menu.py#L93
We could make the method broader for every keyword, such as 'other day', because know we are only using it for the week, month and year.

> Seems like we need to move the "Every" prefix into the strings themselves

Sorry, but I don't understand how that would help. Is having the prefix and the suffix separate makes it harder and not flexible to translate?
Wouldn't that make the parsing 'harder' in `dates.py`?

**Comment by Neui** [comment]
> > (and is "ugly" in the sense that it says "other-day" instead of like "other day")
> 
> Yeah, I know. It is not even easy to write but I had to make it that way as it would not be parsed correctly in the quick add entry otherwise.

Except it is used when displaying it like in the screenshots above, it shouldn't affect parsing code.

> > To allow flexibility for other languages, maybe use more specific strings such as "Every other day", "Every {day}. of the Month", "Every {weekday}", ...)
> 
> We are currently doing similar formatting here. Could you point me to the problem? (I don't know how translating works).
> 
> https://github.com/getting-things-gnome/gtg/blob/024029129c267fbaa82a74660e1ed7cb2d7b98ba/GTG/gtk/editor/recurring_menu.py#L93
> 

I looked at it, I think I didn't get the translation because the file isn't in the `po/POTFILES.in` file.
However, the following line makes it harder to translate when the "Every" part would come after the string:
https://github.com/getting-things-gnome/gtg/blob/024029129c267fbaa82a74660e1ed7cb2d7b98ba/GTG/gtk/editor/recurring_menu.py#L113
My suggestion would be to use something like
```python
            self.title.set_markup(_('Every <b>{formated_term}</b>').format(formated_term=formated_term))
```
(Note that I don't know any language that does this, but better be safe)

> We could make the method broader for every keyword, such as 'other day', because know we are only using it for the week, month and year.

Yes.

> > Seems like we need to move the "Every" prefix into the strings themselves
> 
> Sorry, but I don't understand how that would help. Is having the prefix and the suffix separate makes it harder and not flexible to translate?
> Wouldn't that make the parsing 'harder' in `dates.py`?

Displaying and parsing shouldn't be connected here.
What it means that you would do in the above code something like:
```python
if formatted_term == 'day': # Recurring daily
    self.title.set_markup(_('Every <b>day</b>'))
elif formatted_term == 'other-day': # Recurring every other day
    self.title.set_markup(_('Every <b>other day</b>'))
```
and the translators would have a bit more freedom to translate.

**Comment by Neui** [comment]
Closing since it is fixed by #498
